reddit.js

```diff
@@ -1,5 +1,63 @@
 var reddit = require("./data.json")
+var datas = reddit["data"];
+var dataChild = datas["children"];
+//console.log(dataChild);
+//console.log(dataChild[0].data);
 
 // Write your solutions below
 
-console.log(reddit["data"]);
\ No newline at end of file
+//for each function
+var forE = function(){
+	dataChild.forEach (function(item){
+		console.log(item.data.title + "is the title");
+	});
+}
+
+var perma = function() {
+	var links = [];
+	//this pushes all permalinks into an array
+	dataChild.forEach ( function(perms){
```

Comment:
Here, ```perms``` may be a misleading parameter name. Since ```forEach``` is iterating over each child of the data.children array, 'child' or 'post' may be a better parameter name to use in this case.